models/classes/theme/ThemeService.php

```diff
@@ -102,10 +102,14 @@ public function removeThemeById($themeId)
             return false;
         }
 
-        $themes = $this->getOption(static::OPTION_AVAILABLE);
-        unset($themes[$themeId]);
+        $availableThemes = $this->getOption(static::OPTION_AVAILABLE);
+        foreach ($this->getAllThemes() as $key => $theme) {
+            if ($theme->getId() === $themeId) {
+                unset($availableThemes[$key]);
```

Comment:
No need to iterate further after the key has been identified.
```suggestion
                unset($availableThemes[$key]);

                break;
```

Reply:
In fact, it would be a tad more readable if another method was introduced, fetching a key corresponding to a theme ID.
e.g.

```php
private function findThemeKey(string $id): ?string
{
    foreach ($this->getAllThemes() as $key => $theme) {
        if ($theme->getId() === $id) {
            return $key;
        }
    }

    return null;
}
```

That IMO gives a better understanding of what that loop's purpose is.

Then you can reduce your main unsetting logic to `unset($availableThemes[$this->findThemeKey($themeId)]);`

Reply:
> That IMO gives a better understanding of what that loop's purpose is.
> 
> Then you can reduce your main unsetting logic to `unset($availableThemes[$this->findThemeKey($themeId)]);`

All should be removed, so it would not work.